### ISSUE TITLE
Numpy call interfering with dask?

### ISSUE THREAD

**Issue by erikvansebille** [OceanParcels/parcels#949]
As I was checking out #948, I saw that in [the snippet ](https://github.com/OceanParcels/parcels/blob/master/parcels/field.py#L156-L158)

```python
            # Hack around the fact that NaN and ridiculously large values
            # propagate in SciPy's interpolators
            self.data[np.isnan(self.data)] = 0.
```
we still use `np` as library, even though later on we use `lib = np if isinstance(self.data, np.ndarray) else da`

We may want to check whether this call to `np.nan()` triggers dask arrays to convert to numpy arrays in some cases? @CKehl, any thoughts?

**Comment by CKehl** [comment]
it doesn't have any effect on `self.data`, cause `isnan(...)` just returns a boolean array telling which values are NaN.

2 issues though:
a) using `np.isnan(...)` on a **virtual** dask array is undefined and possibly yield an empty array all the time (at least if the related chunk is not loaded to memory)
b) Dask usually has a problem with 'fancy indexing' (NumPy vocab), and indexing by boolean array is 'fancy indexing'. So, if that array is correctly populated, this may result in an error stating that `Dask does not support fancy indexing`.

If it's easy to replace `np.isnan(...)` with `lib.isnan(...)`, then we should do that.

**Comment by erikvansebille** [comment]
Fixed in #950
